server/bin/pbench-tarball-stats.py

```diff
@@ -0,0 +1,196 @@
+  * Top 10 most recent controllers
+  * Top 10 largest tar balls
+"""
+
+import collections
+import locale
+import os
+import re
+import sys
+
+from argparse import ArgumentParser
+from datetime import datetime
+
+import pbench
+
+
+_NAME_ = "pbench-tarball-stats"
+
+tb_pat = re.compile(
+    r"\S+_(\d\d\d\d)[._-](\d\d)[._-](\d\d)[T_](\d\d)[._:](\d\d)[._:](\d\d)\.tar\.xz"
+)
+
+TarBallInfo = collections.namedtuple("TarBallInfo", ["ctrl", "tb", "dt", "stat"])
+
+
+def gen_tar_balls(archive):
+    """Traverse the given ARCHIVE hierarchy looking tar balls, generating
+    TarBallInfo tuples containing information about the tar balls found.
+
+    The tuple contains: (
+        controller name,
+        tar ball name,
+        datetime derived from the tar ball name (
+            not the file system's modification time stamp),
+        the "stat" object for that file
+    )
+
+    If we find a file in a controller directory that does not match the
+    expected tar ball file name pattern, and is not a tar ball .md5 file,
+    we STILL generated a tuple for that file but with None for the datetime
+    of that tuple.
+    """
+    with os.scandir(archive) as archive_scan:
+        for c_entry in archive_scan:
+            if c_entry.name.startswith(".") or not c_entry.is_dir(
+                follow_symlinks=False
+            ):
+                continue
+            # We have a controller directory.
+            with os.scandir(c_entry.path) as controller_scan:
+                for entry in controller_scan:
+                    if entry.name.startswith(".") or not entry.is_file(
+                        follow_symlinks=False
+                    ):
+                        continue
+                    match = tb_pat.fullmatch(entry.name)
+                    if match:
+                        # Turn the pattern components of the match into a datetime
+                        # object.
+                        tb_dt = datetime(
+                            int(match.group(1)),
+                            int(match.group(2)),
+                            int(match.group(3)),
+                            int(match.group(4)),
+                            int(match.group(5)),
+                            int(match.group(6)),
+                        )
+                    elif entry.name.endswith(".md5"):
+                        continue
+                    else:
+                        tb_dt = None
+                    yield TarBallInfo(
+                        ctrl=c_entry.name, tb=entry.name, dt=tb_dt, stat=entry.stat()
+                    )
+
+
+def report_by_bucket(buckets):
+    for bucket_name, bucket_list in sorted(buckets.items()):
+        count = len(bucket_list)
+        ec2_cnt = 0
+        for item in bucket_list:
+            if item[0].startswith("EC2::"):
+                ec2_cnt += 1
+        pct_ec2 = (ec2_cnt / count) * 100.0
+        print(f"{bucket_name}: {count:7n} (%EC2 {pct_ec2:0.02f})")
```

Comment:
"`%EC2`" is slightly cryptic...I think it would be better if the percent sign followed the number:
```
        print(f"{bucket_name}: {count:7n} (EC2 {pct_ec2:0.02f}%)")
```
or
```
        print(f"{bucket_name}: {count:7n} ({pct_ec2:0.02f}% EC2)")
```